### ISSUE TITLE
Unable to install `mesheryctl` v0.6.5, `meshery-perf` and `meshery-cillium` are crashing

### ISSUE THREAD

**Issue by acald-creator** [meshery/meshery#6178]
#### Current Behavior
Current release `v0.6.5` failed to complete installation, both `meshery-perf` and `meshery-cillium` are crashing. From the previous release, `meshery-perf` and `meshery-cpx` are supposed to be deprecated, however, installing the Helm charts, still pulled in `meshery-perf` even though it is disabled.

For Cillium, here is an existing issue open [#71](https://github.com/meshery/meshery-cilium/issues/71) at the `meshery-cillium` error provided still persist in my cluster.

I have a possible fix but I'm currently testing it at this time, I would like to submit a PR and I am open to anyone willing to help with me.

#### Expected Behavior
Installation is supposed to be successful.

#### Screenshots/Logs
```bash
kubectl get pods -n meshery
NAME                                    READY   STATUS             RESTARTS        AGE
meshery-meshery-cpx-5b49b48fdf-7h4g5    1/1     Running            0               5m45s
meshery-istio-65f67fb978-rtfgr          1/1     Running            0               5m45s
meshery-linkerd-cc65f9d78-fspnl         1/1     Running            0               5m44s
meshery-consul-8484fbcc96-kxkjx         1/1     Running            0               5m45s
meshery-operator-7bb97dfdb-j796z        2/2     Running            0               5m44s
meshery-osm-6fbc9769cd-lcpjs            1/1     Running            0               5m43s
meshery-697fd8446-4zgbz                 1/1     Running            0               5m43s
meshery-app-mesh-67c78f557d-xgww5       1/1     Running            0               5m45s
meshery-kuma-785765997-xfzcc            1/1     Running            0               5m44s
meshery-nsm-7788546ff6-wbzgz            1/1     Running            0               5m44s
meshery-traefik-mesh-c5548d4f6-7wk4t    1/1     Running            0               5m43s
meshery-nginx-sm-5c5569dbf4-dx9kg       1/1     Running            0               5m44s
meshery-broker-0                        2/2     Running            0               5m39s
meshery-meshsync-7b498c5686-dwstd       1/1     Running            2 (5m24s ago)   5m38s
meshery-meshery-perf-5bbd4f86df-89pp8   0/1     CrashLoopBackOff   5 (2m39s ago)   5m43s
meshery-cilium-7757c94c4d-46225         0/1     CrashLoopBackOff   5 (2m11s ago)   5m45s
```

#### Environment

- **Host OS:** Linux and Ubuntu 22 WSL2
- **Platform:** Kubernetes k3d
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

**Comment by acald-creator** [comment]
Removing the two charts: `meshery-perf` and `meshery-cpx` from the `values.yaml` did not work, I tried removing any instances of them being mentioned via the Charts. However, deleting the two folders, solved the issue of keeping the two Docker images being pulled into the Helm deployment.

Please let me know if this was the intention to remove these two from the charts, and not to be used anymore via deployment.

Cilium still persist with the error as before.

**Comment by leecalcote** [comment]
@CyrineG @alphaX86 @AugustasV

**Comment by CyrineG** [comment]
> Removing the two charts: meshery-perf and meshery-cpx from the values.yaml did not work, I tried removing any instances of them being mentioned via the Charts. However, deleting the two folders, solved the issue of keeping the two Docker images being pulled into the Helm deployment.

@acald-creator that is weird. Setting the condition values should be [enough](https://helm.sh/docs/topics/charts/#tags-and-condition-fields-in-dependencies), unless the boolean variable path defined is not the correct one :thinking:

**Comment by acald-creator** [comment]
@CyrineG I'm thinking thee same thing as well, I'll see if I can make some changes to test to see if it works out or not, where I may not have to removed the folders completely.

**Comment by acald-creator** [comment]
I haven't made any changes but was there something changed? This is occurring from the `master` branch and `v0.6.5` tag. I even clone the project separately, and this error came up.

```bash
helm install meshery meshery/ --debug --dry-run
install.go:178: [debug] Original chart version: ""
install.go:199: [debug] CHART PATH: /home/auxyria/go/src/github.com/acald-creator/meshery/install/kubernetes/helm/meshery

install.go:206: [debug] WARNING: This chart or one of its subcharts contains CRDs. Rendering may fail or contain inaccuracies.
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: [unable to recognize "": no matches for kind "Broker" in version "meshery.meshery.io/v1alpha1", unable to recognize "": no matches for kind "MeshSync" in version "meshery.meshery.io/v1alpha1"]
helm.go:88: [debug] [unable to recognize "": no matches for kind "Broker" in version "meshery.meshery.io/v1alpha1", unable to recognize "": no matches for kind "MeshSync" in version "meshery.meshery.io/v1alpha1"]
unable to build kubernetes objects from release manifest
helm.sh/helm/v3/pkg/action.(*Install).RunWithContext
        helm.sh/helm/v3/pkg/action/install.go:277
main.runInstall
        helm.sh/helm/v3/cmd/helm/install.go:265
main.newInstallCmd.func2
        helm.sh/helm/v3/cmd/helm/install.go:125
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.2.1/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.2.1/command.go:974
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.2.1/command.go:902
main.main
        helm.sh/helm/v3/cmd/helm/helm.go:87
runtime.main
        runtime/proc.go:225
runtime.goexit
        runtime/asm_amd64.s:1371
INSTALLATION FAILED
main.newInstallCmd.func2
        helm.sh/helm/v3/cmd/helm/install.go:127
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.2.1/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.2.1/command.go:974
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.2.1/command.go:902
main.main
        helm.sh/helm/v3/cmd/helm/helm.go:87
runtime.main
        runtime/proc.go:225
runtime.goexit
        runtime/asm_amd64.s:1371
````

**Comment by acald-creator** [comment]
@alphaX86 Hi there, the names being appended with `meshery-` that's when I did a local installation trying to test the changes.

However, using the `curl -L https://meshery.io/install | PLATFORM=kubernetes bash -` resulted in this output.

```bash
kubectl get pods -n meshery
NAME                                   READY   STATUS             RESTARTS      AGE
meshery-nsm-7788546ff6-hrhvp           1/1     Running            0             13m
meshery-consul-8484fbcc96-wkbxh        1/1     Running            0             13m
meshery-cpx-7c6885bcfc-h92ld           1/1     Running            0             13m
meshery-operator-7bb97dfdb-f9jp5       2/2     Running            0             13m
meshery-cilium-7757c94c4d-xxgtz        1/1     Running            0             13m
meshery-osm-6fbc9769cd-sq78k           1/1     Running            0             13m
meshery-app-mesh-67c78f557d-rm7jq      1/1     Running            0             13m
meshery-nginx-sm-5c5569dbf4-f5xsj      1/1     Running            0             13m
meshery-linkerd-cc65f9d78-thdxx        1/1     Running            0             13m
meshery-istio-65f67fb978-gh4sg         1/1     Running            0             13m
meshery-broker-0                       2/2     Running            0             12m
meshery-traefik-mesh-c5548d4f6-5g7bt   1/1     Running            0             13m
meshery-kuma-785765997-52wrm           1/1     Running            0             13m
meshery-meshsync-689bb4f698-pdhcl      1/1     Running            1 (11m ago)   12m
meshery-5d8d749fb4-x7zrm               1/1     Running            0             13m
meshery-perf-794bb4f4c7-dccz5          0/1     CrashLoopBackOff   7 (28s ago)   13m
```

But `meshery-perf` and `meshery-cpx` are still being pulled in, even though they are deprecated? Are they meant not to be pulled in the Helm deployment?

**Comment by alphaX86** [comment]
@acald-creator they shouldn't that's why I declared them as deprecated last week to avoid them getting pulled. This is strange... seems they are dependent on something else, time to check it deep

**Comment by alphaX86** [comment]
Oh wait.... the charts are being repeated again... under `meshery-operator` 👀 
![image](https://user-images.githubusercontent.com/59508546/188664896-5a50d440-c47b-46a8-bc57-11ef2289dd34.png)

// @leecalcote

**Comment by acald-creator** [comment]
Yes, that was going to be my next question, because it is repeated in `meshery-operator` and I also had a question about the versioning too, since it is at `0.5.0` and the latest release is `v0.6.5`.

**Comment by acald-creator** [comment]
My initial thought was that the two deprecated charts were been pulled in because it is still in the `meshery-operator` but I wasn't sure how dependent `meshery` is on `meshery-operator`.

**Comment by CyrineG** [comment]
So our issue right now here is that:

1- we can't delete their definition as dependencies in meshery/Chart.yaml while keeping their charts under /charts 
=> because Helm expects all dependencies under /charts to be declared in Chart.yaml else the Chart Release tests fail
2- the meshery-perf and meshery-cpx are still deployed despite disabling them in Values.yaml 
=> I wonder if using tags (like [here](https://helm.sh/docs/topics/charts/#tags-and-condition-fields-in-dependencies)) to disable them instead could work?

**Comment by acald-creator** [comment]
After the PR was reverted, I'm testing to see if keeping them disabled in either `meshery-operator` or `meshery` makes a different. Even declaring them deprecated should do something too, but I'll return with a response.

**Comment by acald-creator** [comment]
After some testing, it seems that `meshery-perf` and `meshery-cpx` takes precedent in the Helm deployment, when it is enabled `true` via `values.yaml` in the `meshery-operator`.

Checking the `meshery` helm locked file, there aren't any `meshery-perf` and `meshery-cpx` but it is still showing through the `meshery-operator` helm locked file.

Just updating to let you know that I'm still testing, because I reached a point even when I enabled false, and removed the charts from the `values.yasml` file, it was still being pulled in.
